lib/liquid/context.rb

```diff
@@ -191,15 +191,12 @@ def evaluate(object)
     def find_variable(key, raise_on_not_found: true)
       # This was changed from find() to find_index() because this is a very hot
       # path and find_index() is optimized in MRI to reduce object allocation
-      index = @scopes.find_index { |s| s.key?(key) }
+      scope = (index = @scopes.find_index { |s| s.key?(key) }) && @scopes[index]
+      scope ||= (index = @environments.find_index { |s| s.key?(key) || !s.default_proc.nil? }) && @environments[index]
```

Comment:
```suggestion
      scope ||= (index = @environments.find_index { |s| s.key?(key) || s.default_proc }) && @environments[index]
```

Slightly smaller bytecode, probably no measurable difference but I also find this quick to understand.

Reply:
Technically calling `key?` on the environment without checking `respond_to` first is a breaking change, but I think that's okay – this seems cleaner.

Reply:
Maybe I'll try a test where `nil` is passed into render if there isn't one already or maybe a Drop that doesn't respond to `key?`

Reply:
OK so I can't pass anything through to environment that isn't either a Hash or a Drop which implements `key?` anyway. So unless a drop is specifically removing the `key?` method this will always work.

Reply:
template has this `raise ArgumentError, "Expected Hash or Liquid::Context as parameter"`